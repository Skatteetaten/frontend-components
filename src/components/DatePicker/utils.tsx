export const DEFAULTFORMATDATE = (date: Date | null | undefined): string => {
  return date
    ? Intl.DateTimeFormat('no', {
        year: 'numeric',
        month: '2-digit',
        day: 'numeric',
      }).format(date)
    : '';
};

export const DEFAULTPARSEDATEFROMSTRING = (date: string): Date | null => {
  try {
    if (date && date.length === 10) {
      const day: number = parseInt(date.substring(0, 2));
      const month: number = parseInt(date.substring(3, 5));
      const year: number = parseInt(date.substring(6, 10));
      const formattedDate: Date = new Date(Date.UTC(year, month - 1, day));
      return formattedDate;
    }
  } catch (error) {
    console.log(error);
  }
  return null;
};

export const monthsForLocale = (
  localeName = 'no',
  monthFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { month: monthFormat })
    .format;
  const months: string[] = [];
  for (let month = 0; month < 12; month++) {
    months.push(format(new Date(Date.UTC(2020, month))));
  }
  return months;
};

export const weekdaysForLocale = (
  localeName = 'no',
  dayFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { weekday: dayFormat })
    .format;
  const weekdays: string[] = [];
  for (let day = 0; day < 7; day++) {
    weekdays.push(format(new Date(Date.UTC(2020, 7, day + 2)))); // add 2 days to let the week start on Sunday
  }
  return weekdays;
};
