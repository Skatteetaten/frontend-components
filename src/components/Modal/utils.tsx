import { useEffect } from 'react';

export const createModalDomPlacement = (
  setFlag: (a: boolean) => void,
  shadowRootNode: Document | ShadowRoot | undefined
) => {
  if (!shadowRootNode) {
    setFlag(true);
    return;
  }

  const styleInjectorWrapperId = 'style-injector';
  const modalWrapperId = 'modal-wrapper';

  if (shadowRootNode.getElementById(modalWrapperId)) {
    setFlag(true);
  } else {
    const modalWrapper = document.createElement('div');
    modalWrapper.id = modalWrapperId;
    const styleInjectorWrapper = shadowRootNode.getElementById(
      styleInjectorWrapperId
    );
    const parent = styleInjectorWrapper ? styleInjectorWrapper : shadowRootNode;
    parent.appendChild(modalWrapper);
    setFlag(true);
  }
};

export const useEscOnPress = (
  ref: HTMLDivElement | null,
  onClose?: (ref: HTMLDivElement) => void
): void => {
  useEffect(() => {
    const onKeyDown = (event: KeyboardEvent): void => {
      if (ref == null) {
        return;
      }
      if (event.code === 'Escape' && onClose) {
        onClose(ref);
      }
    };

    if (ref) {
      window.addEventListener('keydown', onKeyDown as EventListener, false);
    }
    return function cleanup(): void {
      if (window) {
        window.removeEventListener('keydown', onKeyDown as EventListener);
      }
    };
  }, [ref, onClose]);
};
