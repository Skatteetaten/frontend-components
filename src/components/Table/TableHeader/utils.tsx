import { TFunction } from 'react-i18next';

export const getIconNameForTheadCell = (
  isSorted: boolean,
  isSortedAscending: boolean
) => {
  if (isSorted) {
    return isSortedAscending ? 'ArrowDown' : 'ArrowUp';
  }

  return 'ArrowUpDown';
};

export const getAriaLabelForTheadCell = (
  name: string | JSX.Element,
  fieldName: string,
  isSorted: boolean,
  isSortedAscending: boolean,
  t: TFunction<Array<string>>
) => {
  let ariaLabel = fieldName;
  if (typeof name === 'string') {
    ariaLabel = isSorted
      ? isSortedAscending
        ? name.concat(' ', t('table.sorted_ascending'))
        : name.concat(' ', t('table.sorted_descending'))
      : name.concat(' ', t('table.sortable'));
  }

  return ariaLabel;
};
