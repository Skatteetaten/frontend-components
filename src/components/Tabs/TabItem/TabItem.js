import * as React from 'react';
import PropTypes from 'prop-types';

export default class TabItem extends React.PureComponent {
  static propTypes = {
    /** @deprecated Teksten p책 arkfanen */
    linkText: PropTypes.string,
    /** Teksten p책 arkfanen */
    headerText: PropTypes.string,
    /** Unik identifikator av arkfanen */
    itemKey: PropTypes.string,
    /** Ikon p책 arkfanen */
    itemIcon: PropTypes.string,
    /** Teller p책 arkfanen */
    itemCount: PropTypes.number,
    /** Overstyring av stiler */
    className: PropTypes.string
  };
  static defaultProps = {
    itemIcon: undefined
  };

  render() {
    const {
      linkText,
      headerText,
      itemIcon,
      itemCount,
      itemKey,
      children,
      className,
      ...props
    } = this.props;

    return (
      <div
        linkText={linkText}
        headerText={headerText}
        itemIcon={itemIcon}
        itemCount={itemCount}
        itemKey={itemKey}
        {...props}
        className={className}
      ></div>
    );
  }
}
