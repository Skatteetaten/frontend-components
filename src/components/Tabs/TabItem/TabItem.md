**TabItem (Enkeltfane): brukes til å lage en enkelt arkfane med tilhørende innhold inni Tabs-komponenten.**

```js noeditor
import { MessageBar } from '@skatteetaten/frontend-components/MessageBar';

<MessageBar>
  Det finnes ingen egne eksempler for TabItem. Gå til <a href="/#tabs">Tabs</a>{' '}
  for å se eksempler på hvordan denne komponenten brukes.
</MessageBar>;
```
