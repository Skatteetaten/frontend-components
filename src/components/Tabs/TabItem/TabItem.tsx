import * as React from 'react';
import { IPivotItemProps } from 'office-ui-fabric-react/lib-commonjs/Pivot';
import { BaseComponent } from '@uifabric/utilities';

export default class TabItem extends BaseComponent<IPivotItemProps, {}> {
  static defaultProps = {
    itemIcon: undefined
  };

  render() {
    const { children, ...props } = this.props;
    return <div {...props}>{children}</div>;
  }
}
