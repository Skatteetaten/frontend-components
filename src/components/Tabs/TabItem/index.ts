export { TabItem as default } from './TabItem';
export * from './TabItem';
