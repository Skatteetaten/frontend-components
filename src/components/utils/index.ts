export { DefaultPalette as Palette, skeColor, skePalette } from './palette';
export type { PaletteProps, SkeColorProp } from './palette';
export {
  Fonts,
  FontSizes,
  LineHeightSizes,
  FontWeights,
  IconFontSizes,
  createFont,
} from './fonts';
export { omstrukturerFlatJSON, t } from './i18n/i18n';
export { getSrOnlyStyle } from './getSrOnlyStyle';
export { languages } from './i18n/languages';
export type { FontSizeType } from './fonts';
export { getFocusStyle } from './getFocusStyle';
export { Heading } from './Heading';
export { UseScreen } from './ScreenPlugin';
export { default as SkeIcons } from './icons/icons-ske';
export { Animation } from './getAnimationStyles';
export { takeIf } from './helpers';
export { generateId } from './generateId';
export { useHotkeys } from './useHotkeys';
