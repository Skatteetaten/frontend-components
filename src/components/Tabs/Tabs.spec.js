import React from 'react';
import { mount } from 'enzyme';
import { matches } from './../utils/test-utils';
import { Tabs } from '.';
import { TabItem } from './TabItem';

function oppsettFullDOM(props) {
  return mount(
    <Tabs {...props}>
      <TabItem
        headerText="Tilgjengelige oppgaver"
        itemCount={23}
        itemKey="itemKey-1"
      >
        Innhold #1
      </TabItem>
      <TabItem headerText="Mine oppgaver" itemCount={2} itemKey="itemKey-2">
        Innhold #2
      </TabItem>
      <TabItem
        headerText="Andres oppgaver"
        itemIcon="Update"
        itemKey="itemKey-3"
      >
        Innhold #3
      </TabItem>
      <TabItem headerText="Nylig behandlet" itemCount={132} itemKey="itemKey-4">
        Innhold #4
      </TabItem>
    </Tabs>
  );
}

describe('Tabs komponent', () => {
  it('matcher snapshot', () => {
    matches(
      <Tabs>
        <TabItem
          headerText="Tilgjengelige oppgaver"
          itemCount={23}
          itemKey="itemKey-1"
        >
          Innhold #1
        </TabItem>
        <TabItem headerText="Mine oppgaver" itemCount={2} itemKey="itemKey-2">
          Innhold #2
        </TabItem>
        <TabItem
          headerText="Andres oppgaver"
          itemIcon="Update"
          itemKey="itemKey-3"
        >
          Innhold #3
        </TabItem>
      </Tabs>
    );
  });

  it('rendrer Tabs med riktig innhold', () => {
    const wrapper = oppsettFullDOM({});

    const tab1 = wrapper.find('CustomizedActionButton').at(0);
    const tab3 = wrapper.find('CustomizedActionButton').at(2);
    let content = wrapper.find('PivotItem');

    expect(wrapper.find('CustomizedActionButton')).toHaveLength(4);
    expect(tab1.find('.ms-Pivot-text').html()).toContain(
      'Tilgjengelige oppgaver'
    );
    expect(tab1.find('.ms-Pivot-count').html()).toContain(23);
    expect(tab1.prop('className')).toContain('is-selected');
    expect(content.find('div').html()).toContain('Innhold #1');
    expect(tab3.exists('IconBase')).toEqual(true);
  });

  it('oppdaterer innhold nÃ¥r TabItem blir klikket', () => {
    const wrapper = oppsettFullDOM({});

    let tab2 = wrapper
      .find('CustomizedActionButton')
      .at(1)
      .find('button.ms-Button--action');
    let content = wrapper.find('PivotItem');

    tab2.simulate('click');
    tab2 = wrapper
      .find('CustomizedActionButton')
      .at(1)
      .find('button.ms-Button--action');
    content = wrapper.find('PivotItem');

    expect(content.find('div').html()).toContain('Innhold #2');
    expect(tab2.prop('className')).toContain('is-selected');
  });
});
