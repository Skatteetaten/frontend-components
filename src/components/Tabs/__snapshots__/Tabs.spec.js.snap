// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs komponent matcher snapshot 1`] = `
<div
  className="SkeTabs-40"
>
  <div
    className="ms-FocusZone css-48"
    data-focuszone-id="FocusZone1"
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDownCapture={[Function]}
  >
    <div
      className="ms-Pivot ms-Pivot--large ms-Pivot--tabs root-41"
      role="tablist"
    >
      <button
        aria-selected={true}
        className="ms-Button ms-Button--action ms-Button--command ms-Pivot-link is-selected linkIsSelected-49"
        data-content="Tilgjengelige oppgaver (23)"
        data-is-focusable={true}
        id="Pivot0-Tab0"
        name="Tilgjengelige oppgaver"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="tab"
        type="button"
      >
        <span
          className="ms-Button-flexContainer flexContainer-50"
          data-automationid="splitbuttonprimary"
        >
          <span
            className="ms-Pivot-linkContent linkContent-44"
          >
            <span
              className="ms-Pivot-text text-45"
            >
               
              Tilgjengelige oppgaver
            </span>
            <span
              className="ms-Pivot-count count-46"
            >
               (
              23
              )
            </span>
          </span>
        </span>
      </button>
      <button
        aria-selected={false}
        className="ms-Button ms-Button--action ms-Button--command ms-Pivot-link link-57"
        data-content="Mine oppgaver (2)"
        data-is-focusable={true}
        id="Pivot0-Tab1"
        name="Mine oppgaver"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="tab"
        type="button"
      >
        <span
          className="ms-Button-flexContainer flexContainer-50"
          data-automationid="splitbuttonprimary"
        >
          <span
            className="ms-Pivot-linkContent linkContent-44"
          >
            <span
              className="ms-Pivot-text text-45"
            >
               
              Mine oppgaver
            </span>
            <span
              className="ms-Pivot-count count-46"
            >
               (
              2
              )
            </span>
          </span>
        </span>
      </button>
      <button
        aria-selected={false}
        className="ms-Button ms-Button--action ms-Button--command ms-Pivot-link link-57"
        data-content="Andres oppgaver xx"
        data-is-focusable={true}
        id="Pivot0-Tab2"
        name="Andres oppgaver"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="tab"
        type="button"
      >
        <span
          className="ms-Button-flexContainer flexContainer-50"
          data-automationid="splitbuttonprimary"
        >
          <span
            className="ms-Pivot-linkContent linkContent-44"
          >
            <span
              className="ms-Pivot-icon icon-47"
            >
              <i
                aria-hidden={true}
                className="root-59"
                data-icon-name="Update"
              >
                Ôëê
              </i>
            </span>
            <span
              className="ms-Pivot-text text-45"
            >
               
              Andres oppgaver
            </span>
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    aria-labelledby="Pivot0-Tab0"
    role="tabpanel"
  >
    <div>
      Innhold #1
    </div>
  </div>
</div>
`;
